### PR TITLE
Accept article on topic

### PR DESCRIPTION
**Pull request by diegodelvalle**

### CODE REVIEW

File: src/middlewares/events.middleware.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        profilePicture: user.profile.image_24,
+        username: user.name,
+        isAdmin: user.is_admin || user.is_owner,
+      });
+
+      next();
+    }
+  } else {
+    // TODO: replace with error block
+    res.error('team not found');
```
Comment (fcrizul) [inline src/middlewares/events.middleware.js:44]: this we replace with the change of the UI?

File: src/middlewares/user.middleware.js [hunk 2]
```diff
@@ -18,16 +18,13 @@ async function getUser(req, res, next) {
...
       next();
     } else { // add user to the team
-      const slackClient = new WebClient();
-
-      const { user } = await slackClient.users.info({
-        token: req.team.token,
-        user: user_id,
-      });
+      const user = await getUser(req.team.token, user_id);
+      console.log(user);
```
Comment (fcrizul) [inline src/middlewares/user.middleware.js:22]: Could we change it to `debug`?

File: src/controllers/interactive.controller.js [hunk 3]
```diff
@@ -1,48 +1,154 @@
...
 
-  const user_id = payload.user.id;
-  const team_id = payload.team.id;
-  const channel_id = payload.channel.id;
-  
   debug("payload: " + JSON.stringify(payload));
 
+  const { user, team } = req;
+
+  console.log(user, team, payload);
```
Comment (fcrizul) [inline src/controllers/interactive.controller.js:74]: Console.log?

File: src/controllers/interactive.controller.js [hunk 4]
```diff
@@ -1,48 +1,154 @@
...
-            // TODO: 
-            res.success();
-            break;
-          default:
-            // TODO: Cambiar por otro mensaje y/o estructura
-            res.error('Ouch!');
-            return;
-        }
+      const blockActionPromises = payload.actions.map(action => {
+        console.log(action);
```
Comment (fcrizul) [inline src/controllers/interactive.controller.js:79]: console.log?

File: src/middlewares/events.middleware.js (hunk 1 above)
Comment (diegodelvalle) [reply src/middlewares/events.middleware.js:44]: yes!

**Review by fcrizul — APPROVED** [review]

File: src/controllers/admin.controller.js [hunk 5]
```diff
@@ -61,6 +61,11 @@ async function addTopic(req, res) {
   }
 }
 
+// { "type": "block_actions", "user": { "id": "U01QWBBE4SZ", "username": "diego.valle", "name": "diego.valle", "team_id": "T01QSKEL1QW" }, "api_app_id": "A01RYPT4Q72", "token": "yQxF4X45RAmngqbXRKB34Fww", "container": { "type": "message", "message_ts": "1619634961.000200", "channel_id": "D01RC2X5L49", "is_ephemeral": true }, "trigger_id": "2011973345219.1842660681846.b8d4c1067ca94fc07c4c4741cfe39b36", "team": { "id": "T01QSKEL1QW", "domain": "slack-bookshelf" }, "enterprise": null, "is_enterprise_install": false, "channel": { "id": "D01RC2X5L49", "name": "directmessage" }, "state": { "values": { } }, "response_url": "https:\/\/hooks.slack.com\/actions\/T01QSKEL1QW\/2011762010354\/bx8Cc56AX7tX9oNpsT3yB9w7", "actions": [{ "action_id": "W7Mlk", "block_id": "Wnv\/s", "text": { "type": "plain_text", "text": "Approve", "emoji": true }, "value": "approve", "style": "primary", "type": "button", "action_ts": "1619634967.532538" }] }
```
Comment (MarianoAmado) [inline src/controllers/admin.controller.js:64]: Be careful with the token and other secrets :p

File: src/views/events.views.js [hunk 6]
```diff
@@ -0,0 +1,151 @@
+function noApprovalRequests () {
+  return [
+    {
+			"type": "header",
```
Comment (MarianoAmado) [inline src/views/events.views.js:4]: fix indentation :)

**Review by MarianoAmado — APPROVED** [review]

**Pull request merged**